content/tutorial-stochastic-simulations.md

```diff
@@ -0,0 +1,483 @@
+|`integers(low, high=, size=)`| Generates `size` integers in the requested range [low, high) in a uniform distribution|
+|`normal(loc, scale=, size=)`| Generates `size` floats in a normal distribution centered on `loc` with a `scale` standard deviation|
+|`binomial(t, p, size=)`| Generates `size` integers in a bionimial distribution with `t` trials for each value with a probability `p`|
+|`choice(a, size=)`| Selects `size` elements from `a` list, tuple or ndarray with an even probability |
+|`shuffle(a)`| Shuffles items in list, tubple, or ndarray `a` |
+
+```{code-cell} ipython3
+rng = np.random.default_rng(seed=18)
+rng.random(10)
+```
+
+You may have guessed by looking at the returned values that `random()` produces
+values in the 0 $\rightarrow$ 1 range, but what happens if we need values in a
+different range?
+We can modify these values by mutiplying them by a coefficient to increase the
+range and using addition or substration to shift the center of the range.
+For example, if we need random values from -10 $\rightarrow$ +10, we can
+accomplish this by the following.
+
+```{code-cell} ipython3
+20 * rng.random(10) - 10
+```
+
+It's worth noting that the `random()` method returns values from the [0,1) range
+which includes the lower end and excludes the upper end.
+This is close enough to [0,1] for our purposes.
+
++++
+
+## Solving the birthday problem
+
+The goal of the [birthday problem](https://en.wikipedia.org/wiki/Birthday_problem)
+is to determine the probability of multiple children in a classroom of 25
+students having the same birthday.
+For simiplicity, we will ignore leap years (i.e., February 29 birthdays) and
+assume that the students are equilly likely to be born any day of the year.
+One method of determining the answer is to repeately simulate a classroom of 25
+student birthdays by using a random number generator and see how often all
+students in the class have unique birthdays.
+The general steps for this processes are outlined below. 
+
+1. Simulate 25 random birthdays with a random integer generator
+2. Check to see if any of the above birthdays match
+3. Record the outcome if all birthdays are unique
+4. Repeat setps 1-3 a large number of times
+
+To make things easier, we will just generate integers 0 $\rightarrow$ 364
+representating days of the year instead of actual months and days.
+
+```{code-cell} ipython3
+def all_unique(class_size, n_classrooms=100):
+    '''Randomly generates birthdays for groups of kids of a given group size and
+    outputs the percent of time all students have uniquie birthdays. All input
+    and output values are ints or floats.
+    
+    (classroom size, number of classrooms) -> percentage classrooms with all unique birthdays
+    '''
+    
+    all_unique_class = 0  # number of classrooms with students NOT sharing birthdays
+    
+    for classroom in range(n_classrooms):
+        birthdays = rng.integers(0, high=365, size=class_size)
```

Comment:
Overall, I think this tutorial is an opportunity to demonstrate good practice for passing PRNG state to functions that consume pseudorandomness. Namely, each function should take an `rng=None` argument and execute `rng = np.random.default_rng(rng)` before calling any methods. @albertcthomas has [a good article on this](https://albertcthomas.github.io/good-practices-random-number-generators/), though we are [converging on using `rng` as the name for the argument instead of `seed`](https://github.com/scientific-python/specs/pull/180).

I think that's more critical information for writing stochastic simulations than details about any particular `Generator` method call.